File: Parte1/EJEMPLOS PATRONES/Python/Builder.py
import abc


class Manufacturer:

    def __init__(self):
        self._builder = None

    def constuct(self, builder):
        self._builder = builder
        self._builder.buildBase()
        self._builder.buildScreen()
        self._builder.buildButtons()


class PhoneBuilder(metaclass=abc.ABCMeta):

    @abc.abstractmethod
    def buildBase(self):
        pass

    @abc.abstractmethod
    def buildScreen(self):
        pass

    @abc.abstractmethod
    def buildButtons(self):
        pass


class WindowsPhoneBuilder(PhoneBuilder):
    def buildBase(self):
        pass

    def buildScreen(self):
        pass

    def buildButtons(self):
        pass


class AndroidPhoneBuilder(PhoneBuilder):
    def buildBase(self):
        pass

    def buildScreen(self):
        pass

    def buildButtons(self):
        pass

File: Parte1/EJEMPLOS PATRONES/PHP/Builder.php
<?php

class Manufacturer {
    private $builder = null;

    public function __contruct($builder) {
        $this->builder = $builder;
    }
    public function contruct() {
        $this->builder->buildBase();
        $this->builder->buildScreen();
        $this->builder->buildButtons();
    }
}

interface PhoneBuilder {
    public function buildBase();
    public function buildScreen();
    public function buildButtons();
}

class WindowsPhoneBuilder implements PhoneBuilder {

    public function buildBase() {
        
        
    }

    public function buildButtons() {
        
        
        
    }

    public function buildScreen() {
        
        
        
        
        //TODO build method
    }

}
class AndroidBuilder implements PhoneBuilder {

    public function buildBase() {
        
        
        
    }

    public function buildButtons() {
        //TODO build method
        
        
        
    }

    public function buildScreen() {
        //TODO build method
        
        
    }

}
